### PR TITLE
Adding repo for centos7 pkg

### PR DESCRIPTION
**Pull request by MalloZup**
we want to add res7.repo

### CODE REVIEW

**Comment by mcalmer** [comment]
I think this needs a bit more work.

**Comment by MalloZup** [comment]
yes. 🍡 . we need to install upgrade the pkg from repo

File: salt/default/repos.sls [hunk 1]
```diff
@@ -180,6 +180,13 @@ galaxy_key:
     - watch:
       - file: galaxy_key
 
+centos-salt-repo:
```
Comment (moio) [inline salt/default/repos.sls]: This is executed on any red hat os, should be restricted to rhel and centos 7 exclusively. Please add conditions.

File: salt/default/repos.d/SUSE_RES-7.repo [hunk 2]
```diff
@@ -0,0 +1,5 @@
+[SUSE_RES-7.repo]
+name=SUSE Manager 3 RES7-SUSE-Manager-Tools (SUSE_RES-7_Update_standard)
+type=rpm-md
+enabled=1
+baseurl=http://{{ grains.get("package-mirror") | default("download.suse.de", true) }}/ibs/Devel:/Galaxy:/Manager:/Head:/RES7-SUSE-Manager-Tools/SUSE_RES-7_Update_standard/
```
Comment (moio) [inline salt/default/repos.d/SUSE_RES-7.repo]: You need a new mirror.lftp line to support this in package-mirror. Please add it

File: salt/default/repos.sls [hunk 3]
```diff
@@ -179,9 +179,18 @@ galaxy_key:
     - name: rpm --import /tmp/galaxy.key
     - watch:
       - file: galaxy_key
-
+{% if grains['os'] == 'CentOS' %}
```
Comment (MalloZup) [inline salt/default/repos.sls]: 1) added here@moio

File: salt/package-mirror/mirror.lftp [hunk 4]
```diff
@@ -89,3 +89,6 @@ mirror --continue --delete --loop --parallel=3 --verbose=3 -i "repodata/.*" -i "
 # Testsuite tools repo
 mirror --continue --delete --loop --parallel=3 --verbose=3 -i "repodata/.*" -i "x86_64/.*"  -i "noarch/.*" -x ".*\.mirrorlist" /ibs/Devel:/Galaxy:/ruby:/test/SLE_12/ ibs/Devel:/Galaxy:/ruby:/test/SLE_12/
 mirror --continue --delete --loop --parallel=3 --verbose=3 -i "repodata/.*" -i "x86_64/.*"  -i "noarch/.*" -x ".*\.mirrorlist" /ibs/Devel:/Galaxy:/BuildRepo/SLE_12_SP1/ ibs/Devel:/Galaxy:/BuildRepo/SLE_12_SP1/
+
+# Centos 7 update repo
+mirror --continue --delete --loop --parallel=3 --verbose=3 -i "repodata/.*" -i "x86_64/.*"  -i "noarch/.*" -x ".*\.mirrorlist" /ibs/Devel:/Galaxy:/Manager:/Head:/RES7-SUSE-Manager-Tools/SUSE_RES-7_Update_standard/
```
Comment (MalloZup) [inline salt/package-mirror/mirror.lftp]: added package mirror support

File: salt/default/repos.d/SUSE_RES-7.repo (hunk 2 above)
Comment (MalloZup) [inline salt/default/repos.d/SUSE_RES-7.repo]: done

**Comment by MalloZup** [comment]
@moio, thx for review. I updated the pr, with stable and nightly support now.

**Comment by moio** [comment]
@MalloZup I did some changes to uniform those repos to the others in sumaform. Please review them commit-by-commit and feel free to ping if they don't make sense to you.

I would also ask you to test them out at least once on four minions with version `3.0-stable`, `3.0-nightly`, `3.1-stable` and `head` by checking that the installed salt version is the intended one (respectively 2015.8.7, 2015.8.12, 2016.11.3 and 2016.11.3).

**Comment by MalloZup** [comment]
@moio , ok thx for the renaming.

One question to your remark:
``` 3.0-stable, 3.0-nightly, 3.1-stable and head``` this vars are not existing for minions atm.
This need another PR for enabling this.

**Comment by moio** [comment]
They are fully supported and documented:

> You can use the `version` variable to override this and set a particular version of Salt - values taken are the same as the SUSE Manager versions (such as 3-nightly or head)

https://github.com/moio/sumaform/blob/master/README_ADVANCED.md#change-saltsuse-manager-tools-version-in-minion-or-client

Please ping me when you have run the tests requested above.

**Comment by MalloZup** [comment]
**version = "3.0-stable"** 
```
dma-minsles12sp2:~ # rpm -q salt
salt-2015.8.12-33.1.x86_64
dma-minsles12sp2:~ # rpm -q salt-minion
salt-minion-2015.8.12-33.1.x86_64

*** version = head ***
```
dma-minsles12sp1ssh:~ # rpm -q salt
salt-2016.11.3-310.12.1.develHead.x86_64
dma-minsles12sp1ssh:~ # rpm -q salt-minion
salt-minion-2016.11.3-310.12.1.develHead.x86_64
```

File: salt/default/repos.sls [hunk 5]
```diff
@@ -181,24 +181,26 @@ galaxy_key:
       - file: galaxy_key
 
 {% if grains['osmajorrelease'] == '7' %}
-{% if 'head' in grains.get('version', '') or '3.1-stable' in grains.get('version', '') %}
+{% if 'head' in grains['version'] or '3.1-stable' in grains['version'] %}
```
Comment (moio) [inline salt/default/repos.sls]: This is not just clearer syntax, the former does not fail if the version grain is not set while your version does. Please revert to the (more verbose, less legible but more correct) syntax above

File: salt/default/repos.sls [hunk 6]
```diff
@@ -181,24 +181,26 @@ galaxy_key:
...
   file.managed:
     - name: /etc/yum.repos.d/Devel_Galaxy_Manager_3.0_RES-Manager-Tools-7-x86_64.repo
     - source: salt://default/repos.d/Devel_Galaxy_Manager_3.0_RES-Manager-Tools-7-x86_64.repo
     - template: jinja
     - require:
       - cmd: galaxy_key
-{% elif 'stable' in grains.get('version', 'stable') %}
-tools-repo:
+{% endif %}
+{% if '3.0-stable' in grains['version'] or 'head' in grains['version'] or '3.1-stable' in grains['version'] %}
```
Comment (moio) [inline salt/default/repos.sls]: Why is this needed? What fails if you do not enable the base repo?

File: salt/default/repos.d/SUSE_RES-7-Update-develtool.repo [hunk 7]
```diff
@@ -0,0 +1,5 @@
+[SUSE_RES-7.repo]
+name=SUSE Manager 3 RES7-SUSE-Manager-Tools (SUSE_RES-7_Update_standard)
+type=rpm-md
+enabled=1
+baseurl=http://{{ grains.get("package-mirror") | default("download.suse.de", true) }}/ibs/Devel:/Galaxy:/Manager:/Head:/RES7-SUSE-Manager-Tools/SUSE_RES-7_Update_standard/
```
Comment (mcalmer) [inline salt/default/repos.d/SUSE_RES-7-Update-develtool.repo]: This whole file seems to be a duplicate of Devel_Galaxy_Manager_Head_RES-Manager-Tools-7-x86_64

Is there a reason for this?

File: salt/default/repos.d/SUSE_RES-7-Update-tool.repo [hunk 8]
```diff
@@ -0,0 +1,5 @@
+[SUSE_Updates_RES_7-CLIENT-TOOLS_x86_64]
+name=SUSE:Updates:RES:7-CLIENT-TOOLS:x86_64 (update
+type=rpm-md
+enabled=1
+baseurl=http://{{ grains.get("package-mirror") | default("euklid.nue.suse.com", true) }}/mirror/SuSE/build-ncc.suse.de/SUSE/Updates/RES/7-CLIENT-TOOLS/x86_64/update/
```
Comment (mcalmer) [inline salt/default/repos.d/SUSE_RES-7-Update-tool.repo]: Same here: duplicate of SLE-Manager-Tools-RES-7-x86_64

**Comment by MalloZup** [comment]
@moio, i think it failed before because of a typo that i did with nigthly instead of nightly.

second point is that we should add always a tool repo fo client, the other is optional because is update one

**Comment by MalloZup** [comment]
@moio, imho we should always add the base repo and the other 2 ones are if nightly or head. please let me know . the changes you requested are there.

**Comment by moio** [comment]
My comment above is not correct. Correct table below.

|           | 3.0-stable     | 3.0-nightly    | 3.1-stable                   | head                         |
|-----------|----------------|----------------|------------------------------|------------------------------|
| sles11sp3 | 2015.8.12-30.1 | 2015.8.12-47.1 | 2016.11.3-310.12.1.develHead | 2016.11.3-310.12.1.develHead |
| sles11sp4 | 2015.8.12-30.1 | 2015.8.12-47.1 | 2016.11.3-310.12.1.develHead | 2016.11.3-310.12.1.develHead |
| sles12    | 2015.8.12-33.1 | 2015.8.12-49.1 | 2016.11.3-310.12.1.develHead | 2016.11.3-310.12.1.develHead |
| sles12sp1 | 2015.8.12-33.1 | 2015.8.12-49.1 | 2016.11.3-310.12.1.develHead | 2016.11.3-310.12.1.develHead |
| sles12sp2 | 2015.8.12-33.1 | 2015.8.12-49.1 | 2016.11.3-310.12.1.develHead | 2016.11.3-310.12.1.develHead |
| centos7   | 2015.8.12-15.1 | 2015.8.12-47.1 | 2016.11.3-12.1               | 2016.11.3-12.1               |

I double checked those, and centos7 is OK in this PR (but I did find bugs for SLES 11 which I am fixing shortly).

I am not going to add the RES repo by default, because there might be cases in which we want CentOS without RES Salt (eg. open source or EPEL Salt).

**Comment by moio** [comment]
Thanks!!!

**Pull request merged**